bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/servlets/FormMetaDataDataSourceServlet.java

```diff
@@ -131,57 +135,49 @@ protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHtt
         if (config != null) {
             FormMetaDataType type = FormMetaDataType.fromString(getParameter(config, TYPE, request, null));
             String dataModel = getParameter(config, DATA_MODEL, request, "");
-            String fieldType = getParameter(config, FIELD_TYPE, request, "");
             actionTypeDataSource = new SimpleDataSource(getDataSourceResources(
-                request.getResourceResolver(), type, dataModel, fieldType).iterator());
+                request, type, dataModel).iterator());
         }
         request.setAttribute(DataSource.class.getName(), actionTypeDataSource);
     }
 
-    private List<Resource> getDataSourceResources(ResourceResolver resourceResolver, FormMetaDataType type, String dataModel,
-        String fieldType) {
+    private List<Resource> getDataSourceResources(SlingHttpServletRequest request, FormMetaDataType type, String dataModel) {
+        ResourceResolver resourceResolver = request.getResourceResolver();
         List<Resource> resources = new ArrayList<>();
         FormMetaData formMetaData = resourceResolver.adaptTo(FormMetaData.class);
         if (formMetaData != null) {
             Iterator<FormsManager.ComponentDescription> metaDataList = null;
             switch (type) {
                 case FORMATTERS:
-                    metaDataList = formMetaData.getFormatters(fieldType);
-                    Map<String, Object> firstEntry = new HashMap<>();
-                    firstEntry.put("text", "Select");
-                    firstEntry.put("value", "");
-                    ValueMap firstEntryVm = new ValueMapDecorator(firstEntry);
-                    resources.add(new ValueMapResource(resourceResolver, "", JcrConstants.NT_UNSTRUCTURED, firstEntryVm));
-                    while (metaDataList.hasNext()) {
-                        FormsManager.ComponentDescription componentDescription = metaDataList.next();
-                        Resource formatterResource = resourceResolver.getResource(componentDescription.getResourceType());
-                        if (formatterResource != null) {
-                            String path = formatterResource.getPath();
-                            ValueMap formatterResourceValueMap = formatterResource.getValueMap();
-                            Iterator<Map.Entry<String, Object>> it = formatterResourceValueMap.entrySet().iterator();
-                            while (it.hasNext()) {
-                                Map.Entry<String, Object> entry = it.next();
-                                String key = entry.getKey();
-                                // This condition is sufficient for property guideComponentType to get ignored
-                                if (key.startsWith("pattern")) {
+                    Resource contentResource = resourceResolver.getResource((String) request.getAttribute(Value.CONTENTPATH_ATTRIBUTE));
```

Comment:
Can we create a utility to fetch policies so that code could be re-used ?

Reply:
done